// luawl/src/api.js
const fetch = require('cross-fetch').default;

async function Request(Url, Data) {
	const Required = JSON.stringify(Object.assign({token: require('./constants').token}, Data));
  const Response = (await fetch(`https://api.luawl.com/${Url}.php`, {
    method: "POST",
    headers: {"Content-Type": "application/json", "User-Agent": "node-fetch"},
    body: Required
  }))

	return await Response.json()
}

async function GetUser(discord_id) {
	try {
		const whitelist = await Request("getKey", {discord_id});
		if (typeof whitelist === "object") {
			return whitelist;
		} else if (whitelist == "No key found.") {
			return "Key Data Not Found.";
		}
		
	} catch(error) {
		console.log(error);
	}
}

async function BuyerRole() {
	return await Request("getBuyerRole")
}

async function Whitelist(discord_id, trial_hours) {
	return await Request("whitelistUser", {discord_id, trial_hours, isTrial: "1", wl_script: "0"})
}

async function UnWhitelist(discord_id) {
	return await Request("deleteKey", {discord_id})
}

async function ResetHWID(discord_id) {
	return await Request("userHwidReset", {discord_id})
}

async function Blacklist(discord_id) {
	return await Request("createBlacklist", {discord_id})
}

async function UnBlacklist(discord_id) {
	return await Request("removeBlacklist", {discord_id})
}

module.exports = {
	Request,
	GetUser,
	ResetHWID,
	BuyerRole,
	Blacklist,
	UnBlacklist,
	Whitelist,
	UnWhitelist
}
